File: 프로그래머스/lv2/12953. N개의 최소공배수/N개의 최소공배수.java
class Solution {
    public int solution(int[] arr) {
        int answer = arr[0];
          for (int num : arr){
            int gcd = num/cal(Math.min(num,answer), Math.max(num,answer));
            answer *= gcd;
          }
        return answer;
    }
    public int cal(int a,int b){
        if(a % b == 0) return b;
        return cal(b,a % b);
    }
}

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.py
def solution(arr):
    answer = arr[0]
    for num in arr:
        gcd = num/cal(min(num, answer), max(num, answer))
        answer = answer * gcd 
    return answer

def cal(a, b):
    if b == 0:
        return a
    return cal(b, a % b)